props/saber_fett263_buttons.h

```diff
@@ -293,248 +664,3398 @@ class SaberFett263Buttons : public virtual PropBase {
+              break;
+            default:
+              break;
+          }
+          char playtrack[128];
+          RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+          track_player_ = GetFreeWavPlayer();
+          if (track_player_) {
+            track_player_->Play(playtrack);
+          } else {
+            STDOUT.println("No available WAV players.");
+          }
+        }
+#else
+        STDOUT.println("Audio disabled.");
+#endif
+      }
+    }
+  }
+  
+  void StopTrackPlayer() {
+    track_mode_ = PLAYBACK_OFF;
+    if (track_player_) {
+      track_player_->Stop();
+      track_player_.Free();
+    } else {
+      StartOrStopTrack();
+    }
+  }
+   
+  void DetectMenuTurn() {
+     if (menu_ || color_mode_ == CC_COLOR_LIST) {
+       if (millis() - last_rotate_millis_ > 1000) {
+        float a = fusor.angle2() - current_menu_angle_;
+        if (a > M_PI) a-=M_PI*2;
+        if (a < -M_PI) a+=M_PI*2;
+        if (a > twist_menu_ * 2/3) {
+          current_menu_angle_ += twist_menu_;
+          if (current_menu_angle_ > M_PI) current_menu_angle_ -= M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_RIGHT);
+          last_rotate_millis_ = millis();
+        }
+        if (a < -twist_menu_ * 2/3) {
+          current_menu_angle_ -= twist_menu_;
+          if (current_menu_angle_ < M_PI) current_menu_angle_ += M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_LEFT);
+          last_rotate_millis_ = millis();
+        }
+      }
+    }
+  }
+
+#ifdef FETT263_EDIT_MODE_MENU
+#define SUBMENUS 9
+#define COLOR_SUB 15
+#define COLOR_OPTIONS 3
+#define GESTURE_OPTIONS 12
+#define SET_SUBMENUS 6
+#define STYLE_SETTINGS 10
+#endif
+
+  enum MenuType {
+    MENU_TOP,
+    MENU_PRESET,
+    MENU_VOLUME,
+    TRACK_PLAYER,    
+    MENU_REHEARSE,
+#ifdef FETT263_EDIT_MODE_MENU
+    MENU_STYLE,
+    MENU_COLOR,
+    MENU_COLOR_CHANGE,
+    MENU_FONT,
+    MENU_TRACK,
+    MENU_LENGTH,
+    MENU_COPY,
+    MENU_DELETE,
+    MENU_RESET,
+    MENU_EXIT,
+    MENU_COLOR_SUB,
+    MENU_COLOR_MODE,
+    MENU_EFFECT,
+    MENU_RESET_COLOR,
+#if NUM_BLADES > 1
+    MENU_COPY_COLOR,
+    MENU_BLADE_STYLE,
+    MENU_BLADE_COLOR,
+    MENU_BLADE_COPY,
+    MENU_BLADE_SETTING,
+    MENU_BLADE_LENGTH,
+#endif
+    MENU_GESTURE_SUB,
+    MENU_SWINGON,
+    MENU_TWISTON,
+    MENU_THRUSTON,
+    MENU_STABON,
+    MENU_FORCEPUSH,
+    MENU_TWISTOFF,
+    MENU_POWERLOCK,
+    MENU_SETTING_SUB,
+    MENU_STYLE_SETTING_SUB,
+    MENU_STYLE_OPTION,
+    MENU_IGNITION_OPTION,
+    MENU_IGNITION_TIME,
+    MENU_RETRACTION_OPTION,
+    MENU_RETRACTION_TIME,
+    MENU_LOCKUP_POSITION,
+    MENU_DRAG_SIZE,
+    MENU_EMITTER_SIZE,
+    MENU_PREON_OPTION,
+    MENU_PREON_SIZE,
+    MENU_DIM_BLADE,
+    MENU_CLASH_THRESHOLD,
+    MENU_SWINGON_SPEED,
+    MENU_FORCEPUSH_LENGTH,
+    MENU_LOCKUP_DELAY,
+    MENU_CLASH_DETECT,
+    MENU_MAX_CLASH,
+    COLOR_BASE,
```

Comment:
all of these should begin with MENU_ too.

Reply:
updated